In [17]:
import pandas as pd
df= pd.read_excel('/home/alanv/Documents/7/mate/historial_clientes.xlsx')
df.head()

Cliente A  Cliente B
0   1.247098   2.095465
1   0.936653   1.484473
2  -0.186285   1.744021
3   1.494336   1.683304
4   2.557917   2.306799

### Prueba Z
Hipotesis nula h0 = Mu1 = Mu2
Estadistico de prueba para saber si aceptar o rechazar
$Z=\frac{\overline{x}-\overline{y}}

Z =( x.mean-y.mean * (mu1 - mu2) )/ sqrt((sigma1^2/m)+(sigma2^2/n))

In [18]:
df.describe()

Cliente A    Cliente B
count  1101.000000  1101.000000
mean      2.006974     2.123149
std       1.025550     1.017484
min      -1.752928    -0.778317
25%       1.357532     1.403232
50%       1.996068     2.141925
75%       2.706549     2.756875
max       5.556386     5.096496

In [19]:
# Realizar el Z test
from statsmodels.stats.weightstats import ztest as ztest
ztest(df['Cliente A'],df['Cliente B'])

(-2.6683380460572605, 0.007622751887296756)

Se rechaza h0 dado que -2.6683 no se encuentra en la region -1.96,1.96

Ademas de que el p-valor (0.0076) no es mayor a 0.05

In [20]:
# Descarga de datos
# ==============================================================================
import pingouin as pg
url = ('https://raw.githubusercontent.com/JoaquinAmatRodrigo/' +
       'Estadistica-machine-learning-python/master/data/births.csv')
datos= pd.read_csv(url)

peso_smokers= datos.loc[datos.smoke=='smoker','weight']
peso_nonsmokers= datos.loc[datos.smoke=='nonsmoker','weight']

In [21]:
pg.ttest(x=peso_smokers,y=peso_nonsmokers)

T        dof alternative     p-val          CI95%   cohen-d  \
T-test -1.496711  89.277413   two-sided  0.137995  [-0.93, 0.13]  0.268758   

         BF10     power  
T-test  0.512  0.338075